app/controllers/pieces_controller.rb

```diff
@@ -0,0 +1,18 @@
+class PiecesController < ApplicationController
+  def index
+    game_id = request.referrer.last.to_i
```

Comment:
You can pass it a value in a view.  So, in the game show view, something like:

```erb
<script type="text/javascript">
window.GAME_ID = <%= @game.id %>;
</script>
```

Then in your move JS, you can reference `window.GAME_ID`;

Alternatively, as a somewhat more robust pattern, you could pass it the pieces URL as a data attribute:

```erb
<div class="chessboard" data-pieces-url="<%= game_pieces_path(@game) %>">
```

Then it doesn't need to know about / care about the URL structure, it just uses whatever's at that URL to build the update endpoint.

Also your JS code could be moved to `application.js` (or an included file there) and then scoped to the board, so something like:

```javascript
$(function() {
  $('.chessboard').each(function() {
    var $chessboard = $(this);
    var piecesURL = $chessboard.data('pieces-url');
  
    // populates the board with pieces
    $.get(piecesURL).success(function(pieces) {
      // ...
    });

    $chessboard.find(".square").droppable(...);

    function placePiece(piece) {
      // ...
      var piece = $.parseHTML(
        '<div class="piece" data-id='+ piece.id +'>' + pieceHtml + '</div>');
      $chessboard.find('.square#'+squareId).append(piece);
      $chessboard.find('.piece').draggable();
    }

    function movePiece(piece) {
      // ..
      $.post(piecesURL + "/" + piece.data("id"), {
        // ..
      }).success();
    }
});
```

That would find any element with the class `chessboard` and use its data attribute to set up the drag / drop.  Also it would scope the finds to that particular board with `$chessboard.find()`.  We likely won't need this on this app, but doing it that way would even allow for multiple boards on the same page.